### ISSUE TITLE
support for relative imports

### ISSUE THREAD

**Issue by ertw** [palantir/tslint#4096]
### Feature request

**Is your feature request that we implement a new rule?**

No

**Is your feature request related to a problem? Please describe.**

I am using fuse-box, which allows [relative imports](https://fuse-box.org/page/configuration#home-directory).

example:
```
import {MyCoolComponent} from '~/someRelativeDirectory/MyCoolModule';
```

Two rules incorrectly flag this syntax:
```
[tslint] Submodule import paths from this package are disallowed; import from the root instead (no-submodule-imports)
[tslint] Module '~' is not listed as dependency in package.json (no-implicit-dependencies)
```

**Describe the solution you'd like**

tslint will understand realative imports.

**Describe alternatives you've considered**

Disable the rules that trigger this - not a good solution, because these rules can help catch import errors.

**Comment by ertw** [comment]
Resolved issue by properly configuring these rules:
```
"no-implicit-dependencies": [true, ["~"]],
"no-submodule-imports": [true, "~"],
```
